FinSearchUnified/BusinessLogic/Models/FindologicArticleModel.php

```diff
@@ -256,6 +257,17 @@ protected function setVariantOrdernumbers()
                 $this->xmlArticle->addOrdernumber(new Ordernumber($detail->getSupplierNumber()));
             }
         }
+        $mainVariantNumber = $this->baseVariant->getNumber();
+        foreach ($orderNumbers as $index => $orderNumber) {
+            if ($orderNumber == $mainVariantNumber) {
+                unset($orderNumbers[$index]);
+                array_unshift($orderNumbers, $mainVariantNumber);
+                break;
+            }
+        }
+        foreach ($orderNumbers as $orderNumber) {
+            $this->xmlArticle->addOrdernumber(new Ordernumber($orderNumber));
+        }
```

Comment:
Is `$this->baseVariant->getNumber()` always the correct ordernumber?

If that is the case, this should work as well:
```suggestion
        array_unshift($orderNumbers, $mainVariantNumber);

        foreach (array_unique($orderNumbers) as $orderNumber) {
            $this->xmlArticle->addOrdernumber(new Ordernumber($orderNumber));
        }
```

Reply:
> Is `$this->baseVariant->getNumber()` always the correct ordernumber?

From all the testing I have done, I can confirm that.

Reply:
Tobias, thank you for suggestion, just need one more condition, in case that all variant are not main.
Only in that case we need to add `mainVariant/mainDetail` to the variant array.
Without that condition we have a problem `testEmptyValue()` test. But with condition (`in_array()`) all test works as expected.